mp-spdz-rs/benches/ciphertext_ops.rs

```diff
@@ -0,0 +1,155 @@
+//! Benchmarks for ciphertext operations
+
+use ark_mpc::algebra::Scalar;
+use criterion::{criterion_group, criterion_main, BenchmarkId, Criterion, Throughput};
+use mp_spdz_rs::benchmark_helpers::random_plaintext;
+use mp_spdz_rs::fhe::{keys::BGVKeypair, params::BGVParams, plaintext::Plaintext};
+use mp_spdz_rs::TestCurve;
+
+/// Benchmark the time to encrypt and decrypt a plaintext
+fn bench_ciphertext_encrypt_decrypt(c: &mut Criterion) {
+    let mut group = c.benchmark_group("ciphertext-ops");
+    let params = BGVParams::<TestCurve>::new_no_mults();
+    let slots = params.plaintext_slots();
+    let mut keypair = BGVKeypair::gen(&params);
+
+    group.throughput(Throughput::Elements(slots as u64));
+    group.bench_function(BenchmarkId::new("encrypt-decrypt", ""), |b| {
+        b.iter_custom(|n_iters| {
+            let mut total_time = std::time::Duration::default();
+            let mut rng = rand::thread_rng();
+
+            for _ in 0..n_iters {
+                let plaintext = random_plaintext(&params);
+
+                let start = std::time::Instant::now();
+                let ciphertext = keypair.encrypt(&plaintext);
+                let _ = keypair.decrypt(&ciphertext);
+                total_time += start.elapsed();
+            }
+
+            total_time
+        })
+    });
+}
+
+/// Benchmark addition between a ciphertext and a plaintext
+///
+/// This includes only the time to add the two values together
+fn bench_ciphertext_plaintext_addition(c: &mut Criterion) {
+    let mut group = c.benchmark_group("ciphertext-ops");
+    let params = BGVParams::<TestCurve>::new_no_mults();
+    let slots = params.plaintext_slots();
+    let keypair = BGVKeypair::gen(&params);
+
+    group.throughput(Throughput::Elements(slots as u64));
+    group.bench_function(BenchmarkId::new("plaintext-add", ""), |b| {
+        b.iter_custom(|n_iters| {
+            let mut total_time = std::time::Duration::default();
+            let mut rng = rand::thread_rng();
+
+            for _ in 0..n_iters {
+                let mut plaintext = random_plaintext(&params);
+                let ciphertext = keypair.encrypt(&random_plaintext(&params));
+
+                let start = std::time::Instant::now();
+                let _ = &ciphertext + &plaintext;
```

Comment:
do we need to black-box this so it doesn't get compiled out? i remember that being a pattern in other benchmarks

Reply:
we don't have to, won't get compiled out across the FFI

As evidence, these benchmarks take in the hundreds of ms to run; so _something_ expensive is happening